### PR TITLE
Buy album

### PR DESCRIPTION
**Pull request by rodrigo103**
## Summary

Added the complete buyAllbum functionality. And its tests.

## Known Issues

No known issues.

## Trello Card

https://trello.com/c/EGTlywHV

### CODE REVIEW

File: app/services/buyAlbum.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/* eslint-disable curly */
+const { users: User } = require('../models');
```
Comment (mpuricawolox) [inline app/services/buyAlbum.js]: `const { users: User, albums: Album } = require('../models');`

Remember to destructure all attributes from the same `require` call.

File: app/services/buyAlbum.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    albumInTheDB = await Album.createModel({
+      originalAlbumId: receivedAlbum.id,
+      originalUserId: receivedAlbum.userId,
+      title: receivedAlbum.title
+    });
+  }
+
+  const albumsFromUser = await getAlbumsFromUser(userObject.dataValues.username);
+
+  const userHasTheAlbum = albumsFromUser.find(element => element.originalAlbumId === albumId);
```
Comment (mpuricawolox) [inline app/services/buyAlbum.js]: Replace `element` with `album`

File: app/services/buyAlbum.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+const buyAlbumForUser = async (albumId, context) => {
+  const { user } = context;
+  const userObject = await User.getOne({ username: user.username });
+
+  if (!userObject) return badRequest('The user is not logged in!');
+
+  let albumInTheDB = await Album.getOne({ originalAlbumId: albumId });
+
+  if (albumInTheDB === null) {
```
Comment (mpuricawolox) [inline app/services/buyAlbum.js]: You can use `if (!albumInTheDB)`

File: app/services/buyAlbum.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      originalUserId: receivedAlbum.userId,
+      title: receivedAlbum.title
+    });
+  }
+
+  const albumsFromUser = await getAlbumsFromUser(userObject.dataValues.username);
+
+  const userHasTheAlbum = albumsFromUser.find(element => element.originalAlbumId === albumId);
+
+  if (userHasTheAlbum === undefined) {
```
Comment (mpuricawolox) [inline app/services/buyAlbum.js]: You can use `if (!userHasTheAlbum)`

File: app/services/users.js [hunk 5]
```diff
@@ -1,7 +1,8 @@
 /* eslint-disable curly */
 const { encryptPasswordAsync } = require('../helpers/encryption');
 const logger = require('../logger/index');
-const { user: User } = require('../models');
+const { users: User } = require('../models');
```
Comment (mpuricawolox) [inline app/services/users.js]: Replace this:

```
const { users: User } = require('../models');
const { albums: Album } = require('../models');
```

WIth:

```
const { users: User, albums: Album } = require('../models');
```

File: migrations/migrations/create-albums.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
+'use strict';
```
Comment (mpuricawolox) [inline migrations/migrations/create-albums.js]: Delete `'use strict';`

File: migrations/migrations/create-albums.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        field: 'created_at',
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
+        field: 'updated_at',
+        type: Sequelize.DATE
+      },
+      deletedAt: {
+        // allowNull: false,
```
Comment (mpuricawolox) [inline migrations/migrations/create-albums.js]: Delete comment

File: migrations/migrations/create-user.js [hunk 8]
```diff
@@ -31,9 +32,21 @@ module.exports = {
...
+        field: 'created_at',
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
+        field: 'updated_at',
+        type: Sequelize.DATE
+      },
+      deletedAt: {
+        // allowNull: false,
```
Comment (mpuricawolox) [inline migrations/migrations/create-user.js]: Delete comment

File: migrations/migrations/create-userAlbums.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        field: 'created_at',
+        type: Sequelize.DATE
+      },
+      updatedAt: {
+        allowNull: false,
+        field: 'updated_at',
+        type: Sequelize.DATE
+      },
+      deletedAt: {
+        // allowNull: false,
```
Comment (mpuricawolox) [inline migrations/migrations/create-userAlbums.js]: Delete comment

File: test/server.spec.js [hunk 10]
```diff
@@ -2,8 +2,18 @@ const { createTestClient } = require('apollo-server-testing'),
   { ApolloServer } = require('apollo-server'),
   schema = require('../app/graphql');
 
-const { query: _query, mutate } = createTestClient(new ApolloServer({ schema }));
+ApolloServer.prototype.setContext = function setContext(newContext) {
```
Comment (mpuricawolox) [inline test/server.spec.js]: Is it required to use this syntax here? `function setContext`

File: seeders/1_seedUser.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
+'use strict';
```
Comment (mpuricawolox) [inline seeders/1_seedUser.js]: I have two questions about these seeders files:

- Where and how are you using them?
- Can you rename them to have better names? You should not use numbers but use a name that express the contents of the files and how the differ from another user seeder for example.

Comment (rodrigo103) [reply seeders/1_seedUser.js]: They are not needed. I used them for manually testing the code. I deleted them.

**Pull request merged**